### PR TITLE
card component: add details button to the card component

### PR DESCRIPTION
**Pull request by mfanaKoPitori**
Related issues: Link to task on mom: [card_component: add "details" button](https://umuzi-projects.monday.com/boards/1274815998/pulses/1301320895)

## Description:

- **These are the requirements**:
  1. This should be a link that looks like a button.
  1. t should redirect to topics/{topic_id} or projects/{project_id}.
  1. It should link directly to the topic or project progress instance.
  1. In Tilde 1, we link to a page like card/{card_id}. This is a problem because card ids are ephemeral, and the whole card creation mechanism needs to be rethought.

## Screenshots/videos

Screenshot 2024-03-05 214835

The button-styled link is available on all cards for now. For now, the tests are only focused on link project type. we will need to refactor the code for topic-based cards.

## I solemnly swear that:

- [x] My code follows the style guidelines of this project
- [x] I have merged the develop branch into my branch and fixed any merge conflicts
- [x] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have tested new or existing tests and made sure that they pass

### CODE REVIEW

File: backend/backend/settings.py [hunk 1]
```diff
@@ -128,7 +128,7 @@
     "zero_marginal_cost_challenges.apps.ZeroMarginalCostChallengesConfig",
     "automarker.apps.AutomarkerConfig",
     "frontend",
-    "project_review_coordination",
+    # "project_review_coordination",
```
Comment (kingraphaii) [inline backend/backend/settings.py]: DOnt forget to put this back

File: backend/frontend/tests/frontend_test_mixin.py [hunk 2]
```diff
@@ -40,7 +40,7 @@ def reverse_url(
     ):
         """A wrapper around Django's reverse function that returns a full URL"""
         end = reverse(viewname, urlconf, args, kwargs, current_app)
-        return f"{self.live_server_url}/{end}"
+        return f"{self.live_server_url}{end}"
```
Comment (kingraphaii) [inline backend/frontend/tests/frontend_test_mixin.py:43]: Why was this necessary? `is there somwhere we hardcoded  URLs?

**Review by kingraphaii — CHANGES_REQUESTED** [review]

File: backend/frontend/tests/frontend_test_mixin.py (hunk 2 above)
Comment (mfanaKoPitori) [reply backend/frontend/tests/frontend_test_mixin.py:43]: @kingraphaii, I think it is necessary. When I run the tests like this: `python mange.py test,` My test fails when the forward slash is included. I found out that the variable `end` in the method `reverse_url` already comes with a forward slash, so I am trying to remove the extra forward slash.

Screenshot 2024-03-07 103737


Is there something I am missing?

**Review by kingraphaii — APPROVED** [review]

File: backend/frontend/templates/frontend/user/board/partial_card.html [hunk 3]
```diff
@@ -201,6 +201,12 @@ <h5 class="w-[auto] h-[auto] text-black text-base font-bold font-'Poppins' leadi
                   hx-swap="outerHTML">Cancel review request</button>
         {% endif %}
       </div>
+
+      <div class="grid grid-cols-2 grid-auto-rows-1fr gap-x-10 gap-y-10 w-full h-[auto]">
+        <div class="Details-button col-span-1 row-span-1 w-full">
```
Comment (kingraphaii) [inline backend/frontend/templates/frontend/user/board/partial_card.html]: The class naming casing is wrong here, also, are really using that class anywhere?

**Review by kingraphaii — CHANGES_REQUESTED** [review]

**Review by kingraphaii — APPROVED** [review]

**Review by sheenarbw — APPROVED** [review]

**Pull request merged**